AppDB/appscale/datastore/datastore_distributed.py

```diff
@@ -1733,7 +1745,7 @@ def __single_property_query(self, query, filter_info, order_info):
       cursor = appscale_stub_util.ListCursor(query)
       last_result = cursor._GetLastResult()
       startrow = self.__get_start_key(prefix, property_name, direction, 
-        last_result)
+        last_result, query=query)
```

Comment:
Eventually, I would like to cover all the edge cases. We will probably need to refactor the logic of `__get_start_key` to behave more like a filter.